element-template-generator/core/src/main/java/io/camunda/connector/generator/java/ClassBasedTemplateGenerator.java

```diff
@@ -99,9 +98,21 @@ public List<io.camunda.connector.generator.dsl.ElementTemplate> generate(
     var connectorInput = template.inputDataClass();
     var context = TemplateGenerationContextUtil.createContext(connectorDefinition, configuration);
 
-    List<PropertyBuilder> properties =
-        new ArrayList<>(
-            TemplatePropertiesUtil.extractTemplatePropertiesFromType(connectorInput, context));
+    List<PropertyBuilder> properties;
+    if (OutboundConnectorFunction.class.isAssignableFrom(connectorDefinition)
+        || InboundConnectorExecutable.class.isAssignableFrom(connectorDefinition)) {
+      // An outbound connector function or inbound connector executable
+      properties =
+          new ArrayList<>(
+              TemplatePropertiesUtil.extractTemplatePropertiesFromType(connectorInput, context));
+    } else {
+      // Probably an operation annotated connector
```

Comment:
We can also check if it's assignable from `OutboundConnectorProvider` to be sure, right?

Reply:
Yes, this part is a leftover when I tried to implement it without the `OutboundConnectorProvider` marker interface solely on the classpath scanning of `@Operation` using https://github.com/classgraph/classgraph.  It works but seems to add some overhead. I will simplify this as suggested.